pkg/external/machineactuator.go

```diff
@@ -197,19 +209,35 @@ func (c *ExternalClient) waitForJob(jobName string, namespace string, retries in
 								glog.Errorf("failed to delete succeeded job %s: %v", job.Name, err)
 							}
 						}
-						return nil
+						return true, nil
 					}
-					return fmt.Errorf("Job %v failed: %v", job.Name, condition.Message)
+					return false, fmt.Errorf("Job %v failed: %v", job.Name, condition.Message)
 				}
 			}
 		}
 		time.Sleep(5 * time.Second)
 
 		options := metav1.GetOptions{ResourceVersion: job.ObjectMeta.ResourceVersion}
 		job, err = c.kubeclient.BatchV1().Jobs(namespace).Get(job.ObjectMeta.Name, options)
+		if err != nil {
+			return false, err
+		}
+
+		// TODO: when host has OFF state, fence command returns code 2, so I need to catch it
+		// in the future need to refactor fence logic container
```

Comment:
The container called by this actuator? Or the container making Machine API calls?

Reply:
Parsing logging is risky.
While the return codes are stable and covered by the API (any change to them would also break the RHEL cluster product), the logging output is definitely not covered, is different for each agent, and could change at any time.